lib/ansible/plugins/callback/slack_github.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import os
+import subprocess
+from ansible.module_utils.urls import open_url
+from ansible.plugins.callback import CallbackBase
+
+
+class CallbackModule(CallbackBase):
+    """This is an ansible callback plugin that sends full
+    play recap to github gist and sends colored notification
+    to  a Slack channel after play ends.
+
+    This plugin makes use of file `slack.json` in your playbook folder
+    with following contents:
+    ```json
+    {
+        "webhook": "Slack incoming webhook url",
+        "username": "Ansible"
+    }
+    ```
+
+    Output: https://gist.github.com/anonymous/f75e04a115ea86b730e41e2ef03cc5f0
+    """
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'notification'
+    CALLBACK_NAME = 'slack_github'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self, display=None):
+        self.disabled = False
+        super(CallbackModule, self).__init__(display=display)
+        self.github_api = 'https://api.github.com/'
+        self._log = []
+
+    def init_config(self, playbook):
+        self.playbook_name = os.path.basename(playbook._file_name)
+        config_file = os.path.join(
+            playbook._basedir.encode(
+                'ascii', 'ignore'), 'slack.json')
+        if not os.path.isfile(config_file):
+            self._display.warning(
+                'File {0} not found. SlackGithub callback has been disabled'.format(config_file))
+            self.disabled = True
+            return
+        with open(config_file) as config_file:
+            self.config = json.load(config_file)
+        try:
+            cmd = ['git', 'config', '--get', 'user.name']
+            username = subprocess.check_output(
+                cmd, universal_newlines=True).split('\0')
+            self.user = username[0].rstrip()
```

Comment:
user will be a byte string here.  You probably should be using text strings everywhere and transforming into a byte string only if sending it over the wire.  On python3, user will still be a byte string (because it's part of what's returned from subprocess) but string literals and many APIs will be text strings.  Combining those will cause tracebacks.  So it's best to convert that here:

``` python
from ansible.module_utils._text import to_text
[..]
self.user = to_text(username[0].rstrip(), errors='surrogate_or_strict')
```